test/pipe.civet

```diff
@@ -288,6 +288,16 @@ describe "pipe", ->
       array.push(3),array.shift(),array.sort(),array
     """
 
+    testCase """
+      multi-line with assignments
+      ---
+      document.createElement('div')
+      ||> .className = 'civet'
+      ||> .appendChild document.createTextNode 'Civet'
+      ---
+      let ref;(ref = document.createElement('div')).className = 'civet',ref.appendChild(document.createTextNode('Civet')),ref
```

Comment:
Intuitively I would expect something more like

```js
const ref = document.createElement('div');
ref.className = 'civet';
ref.appendChild(document.createTextNode('Civet'));
```

which would also be more TypeScript-friendly. I can't imagine TS likes the isolated `let ref`.

Is there a reason for it being this instead?

Reply:
[TypeScript is happy with it](https://www.typescriptlang.org/play?#code/DYUwLgBATiBmDcAoAFDWEC8EAmB7AxgK4C2IAdmAHT4wCGYIAoqKRcgOTYCWAbuwJT9qwWgGdRAOVqlMEdvl7h2AGkRpKtAA6by2AMIALLsGzI8RVlRoh6IACogAHmAm5sIDnsVgB-VWkQAekCAPQB+IA).

We use this technique a bunch, and this isn't related to this PR, but here's the motivation:

```js
if bool or (document.createElement('div') ||> ...)
↓↓↓
let ref
if(bool || (ref = document.createElement('div'), ..., ref)) {
```
```

We want to short-circuit before doing the `ref` assignment.